dockerfile_parse/parser.py

```diff
@@ -533,6 +581,85 @@ def _add_instruction(self, instruction, value):
             lines += new_line
             self.lines = lines
 
+    def add_lines(self, *lines, **kwargs):
+        """
+        Add lines to the beginning or end of the build.
+        :param lines: one or more lines to add to the content, by default at the end.
+        :param all_stages: bool for whether to add in all stages for a multistage build
+                           or (by default) only the last.
+        :param at_start: adds at the beginning (after FROM) of the stage(s) instead of the end.
+        """
+        assert len(lines) > 0
+        all_stages = kwargs.pop('all_stages', False)
+        at_start = kwargs.pop('at_start', False)
+        assert not kwargs, "Unknown keyword argument(s): {}".format(kwargs.keys())
+
+        froms = [
+            instr for instr in self.structure
+            if instr['instruction'] == 'FROM'
+        ] or [{'endline': -1}]  # no FROM? fake one before the beginning
+        if not all_stages:  # only modify the last
+            froms = [froms[-1]]
+
+        df_lines = self.lines
+        # make sure last line has a newline if lines are to be appended
+        if df_lines and not at_start:
+            df_lines[-1] = _endline(df_lines[-1])
+
+        # iterate through the stages in reverse order
+        # so adding lines doesn't invalidate line numbers from structure dicts.
+        # first add a bogus instruction to represent EOF in our iteration.
+        froms.append({'startline': len(df_lines) + 1})
+        for stage in range(len(froms)-2, -1, -1):  # e.g. 0 for single or 2, 1, 0 for 3 stages
+            start, finish = froms[stage], froms[stage+1]
+            linenum = start['endline'] + 1 if at_start else finish['startline']
+            df_lines[linenum:linenum] = [_endline(line) for line in lines]
```

Comment:
Nit: we're calculating _endline() repeatedly for the same values. Can we move that out of the loop?